mod-source-record-manager-server/src/test/java/org/folio/services/RawRecordsFlowControlServiceImplTest.java

```diff
@@ -0,0 +1,178 @@
+package org.folio.services;
+
+import org.folio.kafka.KafkaConsumerWrapper;
+import org.folio.services.flowcontrol.RawRecordsFlowControlServiceImpl;
+import org.folio.verticle.consumers.consumerstorage.KafkaConsumersStorage;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.util.Collections;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static org.folio.rest.jaxrs.model.DataImportEventTypes.DI_RAW_RECORDS_CHUNK_READ;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RawRecordsFlowControlServiceImplTest {
+
+  @Mock
+  private KafkaConsumersStorage kafkaConsumersStorage;
+
+  @InjectMocks
+  private RawRecordsFlowControlServiceImpl service;
+
+  @Before
+  public void setUp() {
+    ReflectionTestUtils.setField(service, "enableFlowControl", true);
+  }
+
+  @Test
+  public void shouldSkipWhenFlowControlDisabled() {
+    ReflectionTestUtils.setField(service, "enableFlowControl", false);
```

Comment:
nitpick: maybe have package private setters and getters for these fields that are modified by reflection. I think there might be some spring util to set these fields in a @Before method.

Reply:
I am not prefer to introduce setter methods just because we need them in tests, so used 
ReflectionTestUtils library from spring-test project and also we use it in other tests aw well.
Basically for each test we need different input params so I've  not used @Before for it.